keadm/cmd/keadm/app/cmd/debug/check.go

```diff
@@ -347,15 +347,27 @@ func CheckHTTP(url string) error {
 
 func CheckRuntime(runtime string) error {
 	if runtime == common.DefaultRuntime {
-		result, err := util.ExecShellFilter(common.CmdGetStatusDocker)
+		ctx, cancel := context.WithTimeout(context.Background(), util.SystemdServiceTimeout)
+		defer cancel()
+
+		d, err := util.NewSystemdDbus(ctx)
 		if err != nil {
 			return err
 		}
-		if result != "active" {
-			return fmt.Errorf("docker is not running: %s", result)
+
+		results, err := d.ListUnitsByPatternsContext(ctx, []string{"Active"}, []string{"docker"})
+		if err != nil {
+			return err
+		}
+
+		for result := range results {
+			if results[result].ActiveState == "Active" && results[result].Name == "docker" {
+				fmt.Printf("docker is running\n")
+				return nil
+			} else if results[result].ActiveState != "Active" && results[result].Name == "docker" {
+				return fmt.Errorf("docker is not running: %s", results[result].LoadState)
+			}
 		}
```

Comment:
this can also be wrapped as a function, so it can be reused to check whether edgecore is running. in keadm/cmd/keadm/app/cmd/util/common.go line 562

Reply:
The latest commit has a refactor to address this. When calling `isEdgeCoreServiceRunning` previously we were just looking to see if the service was `enabled` with this change we are checking that it is `Active` now, if we dont want this we will need another function to check if they are enabled as well.